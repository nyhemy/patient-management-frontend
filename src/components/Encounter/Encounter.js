import React from 'react';
import { useHistory } from 'react-router-dom';
import styles from './Encounter.module.css';

/**
 * Component which takes data from backend Encounter entity and displays it
 *
 * @param {*} props are props passed to component when it is called
 */
const Encounter = (props) => {
  const {
    id,
    patientId,
    visitCode,
    provider,
    date
  } = props;

  const history = useHistory();

  /**
   * Redirects to EncounterDetails
   */
  const encounterDetails = () => {
    history.push(`/patients/${patientId}/encounters/${id}`);
  };

  return (
    <div className={styles.card}>
      <div>
        <div>
          Id:
          {` ${id}`}
        </div>
        <div>
          Visit Code:
          {` ${visitCode}`}
        </div>
        <div>
          Provider:
          {` ${provider}`}
        </div>
        <div>
          Date:
          {` ${date}`}
        </div>
        <div>
          <button type="button" onClick={encounterDetails}>Details</button>
        </div>
      </div>
    </div>
  );
};

export default Encounter;
