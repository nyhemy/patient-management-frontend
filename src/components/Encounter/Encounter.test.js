import React from 'react';
import { render } from '@testing-library/react';
import { unmountComponentAtNode } from 'react-dom';
import renderer from 'react-test-renderer';
import Encounter from './Encounter';
import '@testing-library/jest-dom/extend-expect';

test('renders w/o crashing', () => {
  const div = document.createElement('div');
  render(<Encounter />, div);
  unmountComponentAtNode(div);
});

test('component matches snapshot', () => {
  const component = renderer.create(<Encounter />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
