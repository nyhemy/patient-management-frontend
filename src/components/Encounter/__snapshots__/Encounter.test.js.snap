// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component matches snapshot 1`] = `
<div
  className="card"
>
  <div>
    <div>
      Id:
       undefined
    </div>
    <div>
      Visit Code:
       undefined
    </div>
    <div>
      Provider:
       undefined
    </div>
    <div>
      Date:
       undefined
    </div>
    <div>
      <button
        onClick={[Function]}
        type="button"
      >
        Details
      </button>
    </div>
  </div>
</div>
`;
